src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs

```diff
@@ -362,14 +363,24 @@ await foreach (var tenant in armClient.GetTenants().GetAllAsync(cancellationToke
             throw new MissingConfigurationException("An azure location/region is required. Set the Azure:Location configuration value.");
         }
 
-        var unique = $"{Environment.MachineName.ToLowerInvariant()}-{environment.ApplicationName.ToLowerInvariant()}";
+        var userSecrets = await userSecretsLazy.Value.ConfigureAwait(false);
+
+        string resourceGroupName;
+        bool createIfAbsent;
 
-        // Name of the resource group to create based on the machine name and application name
-        var (resourceGroupName, createIfAbsent) = _options.ResourceGroup switch
+        if (string.IsNullOrEmpty(_options.ResourceGroup))
         {
-            null or { Length: 0 } => ($"rg-aspire-{unique}", true),
-            string rg => (rg, _options.AllowResourceGroupCreation ?? false)
-        };
+            // Create a unique resource group name and save it in user secrets
+            resourceGroupName = $"rg-aspire-{RandomNumberGenerator.GetHexString(8, true)}-{environment.ApplicationName.ToLowerInvariant()}";
```

Comment:
```suggestion
            resourceGroupName = $"rg-aspire-{environment.ApplicationName.ToLowerInvariant()}-{RandomNumberGenerator.GetHexString(8, true)}";
```

Maybe put the constant thing at the front, so they are sorted together.